### ISSUE TITLE
Docker image errors

### ISSUE THREAD

**Issue by cncoleman** [YaleDHLab/pix-plot#85]
I am getting the following incompatibility errors: 
ERROR: google-auth 1.8.1 has requirement setuptools>=40.3.0, but you'll have setuptools 20.7.0 which is incompatible.
ERROR: markdown 3.1.1 has requirement setuptools>=36, but you'll have setuptools 20.7.0 which is incompatible.
ERROR: tensorboard 2.0.2 has requirement setuptools>=41.0.0, but you'll have setuptools 20.7.0 which is incompatible.
ERROR: tensorflow 2.0.0 has requirement numpy<2.0,>=1.16.0, but you'll have numpy 1.14.3 which is incompatible.

And then when I attempt to process the images, I get:

  File "utils/process_images.py", line 40, in <module>
    flags = tf.app.flags
AttributeError: module 'tensorflow' has no attribute 'app'

**Comment by duhaime** [comment]
@cncoleman Thanks for reaching out! This is kind of interesting--my hunch is that the Docker image you are interacting with was created before https://github.com/YaleDHLab/pix-plot/commit/19947f38f5614b8961988684a5f2b8e19b26f534#diff-399d4153eb7850d9e753a39ce95351db landed. That's the commit that specified the tensorflow version should be <=2.x (e.g. it specified that tensorflow should be 1.x). 

#### Recommended path forward

If I were you, I would destroy the image: `docker rmi pixplot`. Then I'd run a fresh clone and create the docker image again following the steps in the readme.

#### For the brave and true

Alternatively, you could drop a shell into your container (the instance of your pixplot image) by running:

```bash
# create an instance of the pixplot image
docker run -dit \
  -v $(pwd)/output:/pixplot/output \
  -v /Users/my_user/Desktop/my_images:/pixplot/images \
  -p 5000:5000 \
  pixplot

# list all containers and look for the most recent pixplot container
docker ps -a 

# drop a shell into the container
docker exec -it {{ CONTAINER_ID, e.g. c9989785d56c }} /bin/bash
```

Then, once you're inside the container, you could manually update the container:

```bash
# upgrade setuptools
pip install setuptools -U 

# install tensorflow 1.x
pip install tensorflow==1.15.0

# run all the commands manually
cd pixplot && python3.6 utils/process_images.py images/*.jpg && \
  python3.6 -m http.server 5000
```
All of that said, I would just reclone and rebuild the image and container if I were you!

If you give either approach a go, I'd be happy to help with any follow up questions you might have!

**Comment by cncoleman** [comment]
@duhaime Thank you for your help. I was actually using the recent container image but I did as you instructed. "Recommened Path Forward." It threw the same errors. I went into requirements.txt and changed it to tensorflow>=1.8.0,<2.0.0 (e.g. I removed the = so it 2.0 is not an option). That worked much better. The only error I see now is 
ERROR: tensorflow 1.15.0 has requirement numpy<2.0,>=1.16.0, but you'll have numpy 1.14.3 which is incompatible.
At this point perhaps I'll move forward with your suggestion to drop a shell into the container and edit. Will let you know.
